DIRECTOR: add Kids Box quirk to run in desktop mode

mistydemeo opened the pull request:
Kids Box boots to a 320x150 splash screen, then switches to 640x480 for the main content. This causes ScummVM to pick an awkward window size: it selects a screen based around the splash screen, then squashes the main game down into that. See the sample images below.

![scummvm-kidsbox-demo-mac-ja-00002](https://user-images.githubusercontent.com/780485/177015763-2330e083-01ae-415b-a2df-54c0e0e01911.png)
<img width="752" alt="Screen Shot 2022-07-01 at 17 21 50" src="https://user-images.githubusercontent.com/780485/177015786-f4f628c1-1fd6-4b61-851f-11a1c58e854b.png">

@sev suggested implementing a quirk which sets the screen to desktop mode, which I confirmed works well when using the desktop mode. This is the start of that work, but it doesn't quite work yet. I've moved `wmMode` to an instance variable on `DirectorEngine` so it's accessible on `g_director`. I've confirmed that the quirk is evaluated before `_wm` is instantiated, but perhaps I'm scoping this incorrectly and it's assigning to the wrong value.

sev- commented on engines/director/director.cpp:
```diff
@@ -178,12 +174,14 @@ Common::Error DirectorEngine::run() {
 
 	_currentPalette = nullptr;
 
-	wmMode = debugChannelSet(-1, kDebugDesktop) ? wmModeDesktop : wmModeFullscreen;
```
move this to `Director::Director` instead of `_wmMode = 0`. Not remove.

sev- commented on engines/director/director.cpp:
```diff
@@ -178,12 +174,14 @@ Common::Error DirectorEngine::run() {
 	_currentPalette = nullptr;
 
-	wmMode = debugChannelSet(-1, kDebugDesktop) ? wmModeDesktop : wmModeFullscreen;
+	if (debugChannelSet(-1, kDebugDesktop)) {
```
Remove this check and the following assignment. Let it live in the constructor.

sev- reviewed (changes requested):
Please try the suggested changes. It could start working.

Also, please move `gameQuirks()` at the very end of the constructor. This way, the quirks will be able to override the `_centerStage` or `_machineType`

mistydemeo commented:
Hm, no dice. I made the suggested changes. `--debugflags=desktop` works, but the new Kids Box default doesn't.

mistydemeo commented:
Looks like there's a few places in Director that branch on `debugChannelSet(-1, kDebugDesktop)` instead of the value of `wmMode`; I'll see if converting that over helps.

mistydemeo commented:
Fixed. I've added one new commit which replaces a number of checks for `kDebugDesktop` with checks for `Graphics::kWMModeNoDesktop`. Previously, even if `wmMode` indicated that desktop mode was enabled, several of the core checks depended on the debug value being set, which prevented enabling desktop mode via other methods. I've tested, and Kids Box now always boots in desktop mode. Other games are unaffected, and the debug flag still works.

sev- commented:
Thank you!
